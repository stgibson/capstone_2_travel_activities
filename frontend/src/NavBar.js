import React from "react";
import { useSelector } from "react-redux";
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";

/**
 * Component for navigation bar
 * @param {Object} param0 
 * @returns JSX code for rendering navigation bar
 */
const NavBar = ({ logout }) => {
  const username = useSelector(store => store.username);

  return (
    <Navbar bg="light" varian="light">
      <Navbar.Brand href="/">Travel Activities</Navbar.Brand>
      {
        username ? (
          <Nav className="ml-auto">
            <Nav.Link href="/favorite-activities">
              My Favorite Activities
            </Nav.Link>
            <Nav.Link href="/create-plan">Create a New Plan</Nav.Link>
            <Nav.Link href="/plans">My Plans</Nav.Link>
            <Nav.Link href="/" onClick={ logout }>Logout</Nav.Link>
          </Nav>
        ) : (
          <Nav className="ml-auto">
            <Nav.Link href="/signup">Sign Up</Nav.Link>
            <Nav.Link href="/login">Log In</Nav.Link>
          </Nav>
        )
      }
    </Navbar>
  );
};

export default NavBar;