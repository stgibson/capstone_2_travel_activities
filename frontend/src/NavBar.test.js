import React from "react";
import { MemoryRouter } from "react-router-dom";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import NavBar from "./NavBar";
import rootReducer from "./reducers/root";

const store = createStore(rootReducer);

const logout = jest.fn();

it("renders without crashing", () => {
  render(
    <Provider store={ store }>
      <MemoryRouter><NavBar logout={ logout } /></MemoryRouter>
    </Provider>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <Provider store={ store }>
      <MemoryRouter><NavBar logout={ logout } /></MemoryRouter>
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot();
});
