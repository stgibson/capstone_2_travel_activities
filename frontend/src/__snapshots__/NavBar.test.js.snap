// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand navbar-light bg-light"
    varian="light"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Travel Activities
    </a>
    <div
      class="ml-auto navbar-nav"
    >
      <a
        class="nav-link"
        data-rb-event-key="/signup"
        href="/signup"
      >
        Sign Up
      </a>
      <a
        class="nav-link"
        data-rb-event-key="/login"
        href="/login"
      >
        Log In
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
